### PR TITLE
Fix weekly rules looking ahead aggressively

### PR DESCRIPTION
**Pull request by shavz**
### Description

Fix the issue where we would lose a week while aggressively catering for bydays

### Changes

With this change, if the current day counts as part of the iterations, it counts it instead of ignoring and moving forward

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -387,22 +391,23 @@ impl<'a> RRule<'a> {
             .unwrap_or(&chrono_weekday_to_rrule_byday(start_date.weekday()))
             .to_owned();
 
+        let interval: u32 = self.interval.parse().unwrap_or(1);
```
Comment (danielmoi) [inline src/lib.rs]: need the fallback?

File: src/lib.rs [hunk 2]
```diff
@@ -387,22 +391,23 @@ impl<'a> RRule<'a> {
...
+            && start_date_with_intervals.ordinal() > start_date.ordinal();
         let days_to_adjust =
-            self.calculate_weekday_distance(by_day, start_date.weekday(), in_future);
+            self.calculate_weekday_distance(by_day, start_date.weekday(), in_future_day);
         start_date_with_intervals = start_date_with_intervals + Duration::days(days_to_adjust);
 
-        let interval: u32 = self.interval.parse().unwrap_or(1);
         let mut next_date = start_date_with_intervals;
         for _i in 0..interval {
+            if start_date_with_intervals.gt(&start_date) {
```
Comment (danielmoi) [inline src/lib.rs]: just gt? or greater than or equal?

File: src/lib.rs (hunk 1 above)
Comment (shavz) [reply src/lib.rs]: yes, the spec recommends it

File: src/lib.rs (hunk 2 above)
Comment (shavz) [reply src/lib.rs]: hmm good question and not sure:

for example, this basic rule (if we run today):

rrule: FREQ=WEEKLY;INTERVAL=1;COUNT=1;BYHOUR=10;BYMINUTE=30;BYDAY=TU

and is run on 16 April 2019 at 10:30:00 am UTC then technically we should include today as a run day, but depending on further testing, happy to modify

**Pull request merged**